crates/fiber/src/arch/aarch64.S

```diff
@@ -10,12 +10,29 @@
 
 #include "header.h"
 
+#ifndef CFG_TARGET_OS_macos
+    .pushsection .note.gnu.property, "a";
+    .p2align 3;
+    .word 4;
+    .word 16;
+    .word 5;
+    .asciz "GNU";
+    .word 0xc0000000; // GNU_PROPERTY_AARCH64_FEATURE_1_AND
+    .word 4;
+    .word 3; // GNU_PROPERTY_AARCH64_FEATURE_1_BTI | GNU_PROPERTY_AARCH64_FEATURE_1_PAC
+    .word 0;
+    .popsection
+#endif
+
 // fn(top_of_stack(%x0): *mut u8)
 HIDDEN(wasmtime_fiber_switch)
 GLOBL(wasmtime_fiber_switch)
 .p2align 2
 TYPE(wasmtime_fiber_switch)
 FUNCTION(wasmtime_fiber_switch):
+#ifndef CFG_TARGET_OS_macos
+    hint #25 // paciasp
```

Comment:
To confirm, this is because the macos assembler doesn't have support for the `paciasp` instruction natively? (do you know if there's a version that does support it?)

Reply:
Oh wow sorry I completely misread "ifndef CFG_TARGET_OS_macos" as "if macos" which is entirely the opposite of what's happening here. 

Otherwise those makes sense! Although the same question still somewhat applies, do general assemblers not recognize `paciasp` as an instruction? (or is there like some common assembler that only recently got support for such syntax?)

Reply:
At least for `bti`, if a currently supported Ubuntu can't assemble the instruction, IMHO it seems like we should probably avoid it for the time being -- we otherwise don't really have stringent version requirements on the system toolchain AFAIK, just on Rust, and I've seen plenty of old Ubuntu (and other distro) installs in production in various places.

Could we `#define _bti hint #XX` and maybe the PAC instructions too for uniformity?

(I'll defer to Alex on this though if he feels differently, he's far more qualified on toolchain questions generally)

Reply:
Oh I'm mostly just curious I don't think any concrete changes are needed. The `// actual-instr` comments are clear enough to me what it's supposed to be. If there's some assembler that needs the `hint ...` instead of the actual instruction seems fine to leave these in.